test/provider_os_memory_config.cpp

```diff
@@ -0,0 +1,202 @@
+/*
+ *
+ * Copyright (C) 2024 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ *
+ */
+
+#include "base.hpp"
+#include "provider_os_memory_internal.h"
+
+#include <numa.h>
+#include <numaif.h>
+
+struct providerConfigTest : testing::Test {
+    umf_memory_provider_handle_t provider = nullptr;
+    umf_os_memory_provider_params_t params = {
+        /* .protection = */ UMF_PROTECTION_READ | UMF_PROTECTION_WRITE,
+        /* .visibility = */ UMF_VISIBILITY_PRIVATE,
+
+        // NUMA config
+        /* .nodemask = */ NULL,
+        /* .maxnode = */ 0,
+        /* .numa_mode = */ UMF_NUMA_MODE_DEFAULT,
+        /* .numa_flags = */ 0,
+
+        // others
+        /* .traces = */ 1,
+    };
+    const size_t size = 128;
+    void *ptr = nullptr;
+    std::string dest = "destination";
+
+    void SetUp() override {
+        int ret = numa_available();
+        if (ret) {
+            GTEST_SKIP() << "Test skipped, NUMA not available";
+        }
+    }
+
+    void TearDown() override {
+        if (ptr) {
+            umfMemoryProviderFree(provider, ptr, size);
+        }
+        if (provider) {
+            umfMemoryProviderDestroy(provider);
+        }
+    }
+
+    void create_provider(umf_os_memory_provider_params_t *params) {
+        auto res = umfMemoryProviderCreate(&UMF_OS_MEMORY_PROVIDER_OPS, params,
+                                           &provider);
+        ASSERT_EQ(res, UMF_RESULT_SUCCESS);
+        ASSERT_NE(provider, nullptr);
+    }
+
+    void allocate_memory() {
+        auto res =
+            umfMemoryProviderAlloc(provider, size, os_get_page_size(), &ptr);
+        ASSERT_EQ(res, UMF_RESULT_SUCCESS);
+        ASSERT_NE(ptr, nullptr);
+    }
+
+    void read_memory() {
+        dest.reserve(size);
+        memcpy(dest.data(), ptr, size);
+        for (int i = 0; i < size; ++i) {
+            EXPECT_EQ(*(static_cast<char *>(ptr) + i), *(dest.data() + i));
+        }
+    }
+
+    void write_memory(std::string_view str) {
+        memset(ptr, '\0', size);
+        str.copy((char *)ptr, str.size());
+        EXPECT_EQ(std::string_view((char *)ptr), str);
```

Comment:
I think you should initialize the string_view using 'size' as well. Otherwise, the view will contain 0 elements (the ctor that takes only a ptr, stops after finding a NULL terminator).

Reply:
But the ptr's data should already begin with the string passed to this function after `str.copy()`. And the string will be null terminated. I think it's sufficient to check just the beginning of the allocated memory to see if write is even possible on this page.

Reply:
Of course, you are right.